from PyQt5 import QtCore, QtGui, QtWidgets
import sqlite3
from PyQt5.QtWidgets import QMessageBox

class Ui_agentsearch(object):
    def setupUi(self, agentsearch):
        agentsearch.setObjectName("agentsearch")
        agentsearch.resize(1352, 837)
        self.centralwidget = QtWidgets.QWidget(agentsearch)
        self.centralwidget.setObjectName("centralwidget")
        self.F1 = QtWidgets.QFrame(self.centralwidget)
        self.F1.setGeometry(QtCore.QRect(-30, 0, 1368, 878))
        self.F1.setMinimumSize(QtCore.QSize(1368, 878))
        self.F1.setMaximumSize(QtCore.QSize(1368, 878))
        self.F1.setStyleSheet("background-color: rgb(169, 173, 255);\n"
"border-radius:20px;")
        self.F1.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.F1.setFrameShadow(QtWidgets.QFrame.Raised)
        self.F1.setObjectName("F1")
        self.F2 = QtWidgets.QFrame(self.F1)
        self.F2.setGeometry(QtCore.QRect(0, 0, 241, 821))
        self.F2.setStyleSheet("background-color: rgb(255, 222, 210);\n"
"border-radius:20px;")
        self.F2.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.F2.setFrameShadow(QtWidgets.QFrame.Raised)
        self.F2.setObjectName("F2")
        self.ACCT = QtWidgets.QRadioButton(self.F1)
        self.ACCT.setGeometry(QtCore.QRect(670, 80, 121, 20))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.ACCT.setFont(font)
        self.ACCT.setObjectName("ACCT")
        self.mob = QtWidgets.QRadioButton(self.F1)
        self.mob.setGeometry(QtCore.QRect(800, 80, 95, 20))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.mob.setFont(font)
        self.mob.setObjectName("mob")
        self.b1 = QtWidgets.QPushButton(self.F1)
        self.b1.setGeometry(QtCore.QRect(650, 190, 171, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.b1.setFont(font)
        self.b1.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.b1.setStyleSheet("background-color: rgb(198, 198, 198);")
        self.b1.setObjectName("b1")
        self.label = QtWidgets.QLabel(self.F1)
        self.label.setGeometry(QtCore.QRect(420, 250, 151, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.F1)
        self.label_2.setGeometry(QtCore.QRect(420, 310, 141, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.label_3 = QtWidgets.QLabel(self.F1)
        self.label_3.setGeometry(QtCore.QRect(420, 370, 141, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.label_4 = QtWidgets.QLabel(self.F1)
        self.label_4.setGeometry(QtCore.QRect(420, 430, 91, 21))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.label_5 = QtWidgets.QLabel(self.F1)
        self.label_5.setGeometry(QtCore.QRect(420, 540, 131, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.label_6 = QtWidgets.QLabel(self.F1)
        self.label_6.setGeometry(QtCore.QRect(420, 590, 91, 31))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(self.F1)
        self.label_7.setGeometry(QtCore.QRect(420, 480, 101, 41))
        font = QtGui.QFont()
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.label_7.setFont(font)
        self.label_7.setObjectName("label_7")
        self.label_8 = QtWidgets.QLabel(self.F1)
        self.label_8.setGeometry(QtCore.QRect(520, 70, 131, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_8.setFont(font)
        self.label_8.setStyleSheet("\n"
"background-color:rgb(194, 247, 255);")
        self.label_8.setObjectName("label_8")
        self.search = QtWidgets.QLineEdit(self.F1)
        self.search.setGeometry(QtCore.QRect(540, 130, 371, 41))
        font = QtGui.QFont()
        font.setPointSize(11)
        font.setBold(True)
        font.setWeight(75)
        self.search.setFont(font)
        self.search.setStyleSheet("background-color: rgb(194, 247, 255)")
        self.search.setText("")
        self.search.setAlignment(QtCore.Qt.AlignCenter)
        self.search.setObjectName("search")
        self.nam = QtWidgets.QLabel(self.F1)
        self.nam.setGeometry(QtCore.QRect(530, 250, 361, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.nam.setFont(font)
        self.nam.setStyleSheet("\n"
"background-color: rgb(194, 247, 255);")
        self.nam.setText("")
        self.nam.setObjectName("nam")
        self.ph = QtWidgets.QLabel(self.F1)
        self.ph.setGeometry(QtCore.QRect(530, 300, 361, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.ph.setFont(font)
        self.ph.setStyleSheet("\n"
"background-color: rgb(194, 247, 255)\n"
";")
        self.ph.setText("")
        self.ph.setObjectName("ph")
        self.em = QtWidgets.QLabel(self.F1)
        self.em.setGeometry(QtCore.QRect(530, 360, 361, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.em.setFont(font)
        self.em.setStyleSheet("\n"
"background-color: rgb(194, 247, 255);")
        self.em.setText("")
        self.em.setObjectName("em")
        self.add = QtWidgets.QLabel(self.F1)
        self.add.setGeometry(QtCore.QRect(530, 420, 361, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.add.setFont(font)
        self.add.setStyleSheet("\n"
"background-color: rgb(194, 247, 255);")
        self.add.setText("")
        self.add.setObjectName("add")
        self.appC = QtWidgets.QLabel(self.F1)
        self.appC.setGeometry(QtCore.QRect(530, 480, 361, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.appC.setFont(font)
        self.appC.setStyleSheet("\n"
"background-color:rgb(194, 247, 255)")
        self.appC.setText("")
        self.appC.setObjectName("appC")
        self.appB = QtWidgets.QLabel(self.F1)
        self.appB.setGeometry(QtCore.QRect(530, 530, 361, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.appB.setFont(font)
        self.appB.setStyleSheet("\n"
"background-color: rgb(194, 247, 255);")
        self.appB.setText("")
        self.appB.setObjectName("appB")
        self.label_9 = QtWidgets.QLabel(self.F1)
        self.label_9.setGeometry(QtCore.QRect(640, 10, 211, 51))
        font = QtGui.QFont()
        font.setPointSize(13)
        font.setBold(True)
        font.setWeight(75)
        self.label_9.setFont(font)
        self.label_9.setStyleSheet("background:rgb(255, 221, 255)")
        self.label_9.setAlignment(QtCore.Qt.AlignCenter)
        self.label_9.setObjectName("label_9")
        self.gen = QtWidgets.QLabel(self.F1)
        self.gen.setGeometry(QtCore.QRect(530, 590, 161, 31))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.gen.setFont(font)
        self.gen.setStyleSheet("\n"
"background-color: rgb(194, 247, 255)")
        self.gen.setText("")
        self.gen.setObjectName("gen")
        agentsearch.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(agentsearch)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1352, 27))
        self.menubar.setObjectName("menubar")
        agentsearch.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(agentsearch)
        self.statusbar.setObjectName("statusbar")
        agentsearch.setStatusBar(self.statusbar)

        self.b1.clicked.connect(self.sButton)

        self.retranslateUi(agentsearch)
        QtCore.QMetaObject.connectSlotsByName(agentsearch)

    def retranslateUi(self, agentsearch):
        _translate = QtCore.QCoreApplication.translate
        agentsearch.setWindowTitle(_translate("agentsearch", "Agent Details"))
        self.ACCT.setText(_translate("agentsearch", "AGENT ID"))
        self.mob.setText(_translate("agentsearch", "MOBILE NO"))
        self.b1.setText(_translate("agentsearch", "SEARCH"))
        self.label.setText(_translate("agentsearch", "NAME"))
        self.label_2.setText(_translate("agentsearch", "PHONE NO"))
        self.label_3.setText(_translate("agentsearch", "EMAIL"))
        self.label_4.setText(_translate("agentsearch", "ADDRESS"))
        self.label_5.setText(_translate("agentsearch", "APPOINTED\n"
" BANK"))
        self.label_6.setText(_translate("agentsearch", "GENDER:"))
        self.label_7.setText(_translate("agentsearch", "APPOINTED\n"
" CUSTOMER"))
        self.label_8.setText(_translate("agentsearch", "SEARCH TYPE"))
        self.search.setPlaceholderText(_translate("agentsearch", "Enter Search Value"))
        self.label_9.setText(_translate("agentsearch", "SEARCH AGENT"))


    def sButton(self):
        sVal = self.search.text()
        con = sqlite3.connect("UMRS.db")
        cur = con.cursor()

        if self.ACCT.isChecked():    
        #     Stype = "Account"
            q = "select AgentName, Email, PhoneNo, Address, Bank, Gender from AgentReg where AID = ?"
            cur.execute(q,(sVal,))
            res = cur.fetchall()

            if res == []:
                msg = QMessageBox()
                msg.setIcon(QMessageBox.Critical)
                msg.setText("NO data found")
                msg.setWindowTitle("Warning")
                msg.show()
                msg.exec_()

            else:
                # print(res)
                newList = []
                for i in res:
                        for element in i:
                                newList.append(element)
                
                self.nam.setText(newList[0])
                self.em.setText(newList[1])
                self.ph.setText(str(newList[2]))
                self.add.setText(newList[3])
                self.appB.setText(newList[4])
                self.gen.setText(newList[5])
        
        if self.mob.isChecked():    
        #     Stype = "Account"
            q = "select AgentName, Email, PhoneNo, Address,Bank, Gender from AgentReg where PhoneNo = ?"
            cur.execute(q,(sVal,))
            res = cur.fetchall()

            if res == []:
                msg = QMessageBox()
                msg.setIcon(QMessageBox.Critical)
                msg.setText("NO data found")
                msg.setWindowTitle("Warning")
                msg.show()
                msg.exec_()

            else:
                # print(res)
                newList = []
                for i in res:
                        for element in i:
                                newList.append(element)
                
                self.nam.setText(newList[0])
                self.em.setText(newList[1])
                self.ph.setText(str(newList[2]))
                self.add.setText(newList[3])
                self.appB.setText(newList[4])
                self.gen.setText(newList[5])


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    agentsearch = QtWidgets.QMainWindow()
    ui = Ui_agentsearch()
    ui.setupUi(agentsearch)
    agentsearch.show()
    sys.exit(app.exec_())
